src/acl/acl.ts

```diff
@@ -142,20 +142,17 @@ export function internal_getContainerPath(resourcePath: string): string {
 }
 
 /**
- * Verify whether an ACL was found for the given Resource.
+ * Verify whether the given Resource has a resource ACL attached.
  *
- * A Resource fetched with its ACL (e.g. using [[getSolidDatasetWithAcl]]) _might_ have a resource ACL attached, but
- * we cannot be sure: it might be that none exists for this specific Resource (in which case the
- * fallback ACL applies), or the currently authenticated user (if any) might not have Control access
- * to the fetched Resource.
+ * A Resource returned using [[getSolidDatasetWithAcl]], i.e., the returned `dataset,
+ * has its resource ACL attached if its resource ACL exists and the current user
```

Comment:
Should I clarify that when getSolidDatasetWithAcl - must also be called with the Control access in case two different users call getSolidDatasetWithAcl and hasResourceAcl?

Reply:
`getSolidDatasetWithAcl` may be called without control access, but then the ACL won't be available to the caller. I think it's one of the tricky parts of the code, so I think you can clarify that as often as possible :)

Reply:
When I get to the getSolidDatasetWithAcl method, I'll go into deeper details there - and then maybe loop back in here, once we're good with how we want to describe getSolidDatasetWithAcl